include/xtl/xthreadpool.hpp

```diff
@@ -0,0 +1,286 @@
+
+            <b>Note:</b> This function may return 0, which means that multi-threading
+            shall be switched off entirely. If an algorithm receives this value,
+            it should revert to a sequential implementation. In contrast, if
+            <tt>numThread() == 1</tt>, the parallel algorithm version shall be
+            executed with a single thread.
+        */
+    int n_worker_threads() const
+    {
+        return numThreads_;
+    }
+
+        /** \brief Get desired number of threads.
+
+            In contrast to <tt>numThread()</tt>, this will always return a value <tt>>=1</tt>.
+        */
+    int n_threads() const
+    {
+        return std::max(1,numThreads_);
+    }
+
+
+  private:
+    // helper function to compute the actual number of threads
+    static int n_worker_threads(const int desired_n_threads)
+    {
+        #ifdef xtl_SINGLE_THREADED
+            return 0;
+        #else
+            return  desired_n_threads >= 0  ? 
+                        desired_n_threads : 
+                        desired_n_threads == Nice ? 
+                            int(std::thread::hardware_concurrency()) / 2 : 
+                            int(std::thread::hardware_concurrency());
+        #endif
+    }
+
+    int numThreads_;
+};
+
+/********************************************************/
+/*                                                      */
+/*                      xthreadpool                      */
+/*                                                      */
+/********************************************************/
+
+    /**\brief Thread pool class to manage a set of parallel workers.
+
+        <b>\#include</b> \<xtl/xthreadpool.hpp\><br>
+        Namespace: xtl
+    */
+class xthreadpool
+{
+  public:
+
+    /** Create a thread pool from xparallel_options. The constructor just launches
+        the desired number of workers. If the number of threads is zero,
+        no workers are started, and all tasks will be executed in synchronously
+        in the present thread.
+     */
+    xthreadpool(const xparallel_options & options = xparallel_options())
+    :   options_(options),
+        stop(false)        
+    {
+        init();
+    }
+
+
+
+    /** Create a thread pool with n threads. The constructor just launches
+        the desired number of workers. If \arg n is <tt>xparallel_options::Auto</tt>,
+        the number of threads is determined by <tt>std::thread::hardware_concurrency()</tt>.
+        <tt>xparallel_options::Nice</tt> will create half as many threads.
+        If <tt>n = 0</tt>, no workers are started, and all tasks will be executed
+        synchronously in the present thread. If the preprocessor flag
+        <tt>xtl_SINGLE_THREADED</tt> is defined, the number of threads is always set
+        to zero (i.e. synchronous execution), regardless of the value of \arg n. This
+        is useful for debugging.
+     */
+    xthreadpool(const int n)
```

Comment:
`const` is unecessary here; the number of threads should be an unsigned integer.

Reply:
I know that it is unnecessary from a technical point of view, but I personally like the const.
Seems like I am not the only one https://stackoverflow.com/questions/117293/use-of-const-for-function-parameters (of course not wanting the const has also big support).
But if you prefer to not have them I can get ride of them